src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,47 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+
+public class Contact {
+	public final String value;
+	private boolean isPrivate;
+	
+	public Contact(String contact, boolean isPrivate, String errorMessage, String validRegex) throws IllegalValueException {
```

Comment:
Interesting design by capturing both the error message and regex in the parameters.